editor/editor_properties.cpp

```diff
@@ -2183,7 +2183,55 @@ void EditorPropertyResource::_menu_option(int p_which) {
 		case OBJ_MENU_NEW_SCRIPT: {
 
 			if (Object::cast_to<Node>(get_edited_object())) {
-				EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				if (get_edited_property() == "script") {
+					EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				} else {
+					// Copied from editor/scene_tree_dock.cpp#L380 and edited for our purposes
```

Comment:
Best reference the actual method instead of a line number, this file changes often so this will be a wrong reference in a matter of weeks.

Reply:
This and the other line reference are here for while I'm working on the PR. I planned to remove these notes after work on the PR was finished and it's confirmed that it's good for merging -- after all, I still have yet to merge the commits.